_posts/en/newsletters/2023-04-19-newsletter.md

```diff
@@ -0,0 +1,206 @@
+      state updates.  She also ensures that the output in the PSBT
+      commits to giving her 100 tokens (and is otherwise correct).  She
+      provides her signature and broadcasts the transaction.
+
+  Although each of the example transfers above was made onchain, it's
+  straightforward to modify the protocol to operate offchain.  Carol
+  gives a copy of the contract to Dan along with the history of state
+  updates leading to her receiving 100 tokens.  She and Dan then
+  coordinate to create an output which receives the 100 tokens and that
+  requires signatures from both of them to spend.  Offchain, they
+  transfer the tokens back and forth by generating many different
+  versions of a transaction that spends the multisignature output, with
+  each offchain spend committing to the distribution of tokens and the
+  identifiers of the outputs which will receive those tokens.  Finally,
+  one of them broadcasts one of spending transactions, putting the state
+  onchain.
+
+  The outputs to which the tokens were assigned may be encumbered by a
+  Bitcoin script that determines who will ultimately control the tokens.
+  For example, they may pay an [HTLC][topic htlc] script that gives
+  Carol the ability to spend the tokens any time if she can provide a
+  preimage and her signature, or which gives Dan the ability to spend
+  the tokens with just his signature after a time lock expires.  This
+  allows the tokens to be used in forwarded offchain payments, such as
+  those used in LN.
+
+  In a [reply][tenga rgb] to the thread, Federico Tenga linked to a
+  RGB-based [LN node][rgb-lightning-sample] based on a fork of [LDK][ldk
+  repo] and that project's [LDK sample][ldk-sample] node.  Following
+  links in that project, we found useful [additional
+  information][rgb.info ln] about LN compatibility.  More information
+  about the RGB protocol may be found on a [website][rgb.tech] hosted by
+  the LNP/BP Association.
+
+  In this week's post, Orlovsky announced the release of RGB v0.10.
+  Most significantly, the new version is not compatible with contracts
+  created for previous versions but is designed to allow all new
+  contracts to be upgraded over time for future changes in the protocol.
+  A number of other improvements have also been implemented, and a
+  roadmap for adding additional features is presented.
+
+  As of this writing, the announcement had received a small amount of
+  discussion on the mailing list.
+
+## Changes to services and client software
+
+*In this monthly feature, we highlight interesting updates to Bitcoin
+wallets and services.*
+
+- **Descriptor wallet library adds block explorer:**
+  [Descriptor wallet library][] is a rust descriptor-based wallet library that
+  builds upon rust-bitcoin and supports [miniscript][topic
+  miniscript], [descriptors][topic descriptors], [PSBTs][topic psbt], and in
+  recent [releases][Descriptor Wallet v0.9.2], a text-based [block explorer][topic block explorers].
```

Comment:
```suggestion
  recent [releases][Descriptor Wallet v0.9.2], a text-based [block explorer][topic block explorers].
  Its key feature is the ability to parse and display extended details on Taproot control blocks from 
  transaction inputs' witnesses, as well as present descriptors and miniscripts matching the
  transaction scripts.
```

Reply:
Took this suggestion with minor rewording and added a helpful link to control blocks for those not familiar.